### PR TITLE
remove the need to use 'call' on Groovy closures in Query Strings, fixes #577

### LINKED ISSUE #577: remove the need to use the 'call' keyword when calling a Groovy closure in the query strings
Currently if you want to call a Groovy closure in the query strings, you must use the 'call' keyword after the closure name, as illustrated below:
```
A = 0;
getPriceClosure = { return A++; }

t = emptyTable(10)
t2 = t.update("X = getPriceClosure.call()").update("Y = A")
```

The use of 'call' is unnatural, forced, and thus user unfriendly.

### PR DESCRIPTION
**Pull request by jmao-denver**
a follow-up enhancement on the heel of #482 for Groovy closure calls

### CODE REVIEW

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java [hunk 1]
```diff
@@ -356,15 +356,16 @@ private Method getMethod(final Class scope, final String methodName, final Class
...
+                for (Method method : classImport.getDeclaredMethods()) {
                     possiblyAddExecutable(acceptableMethods, method, methodName, paramTypes, parameterizedTypes);
                 }
             }
-            // for Python func call syntax without the explicit 'call' keyword, check if it is defined in Query scope
+            // for Python function/Groovy closure call syntax without the explicit 'call' keyword, check if it is defined in Query scope
             if (acceptableMethods.size() == 0) {
-                if ( variables.get(methodName) == PythonScopeJpyImpl.CallableWrapper.class) {
-                    for (Method method : PythonScopeJpyImpl.CallableWrapper.class.getDeclaredMethods()) {
+                Class methodCls = variables.get(methodName);
```
Comment (kosak) [inline DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:366]: "final Class methodClass..." :-)

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java [hunk 2]
```diff
@@ -356,15 +356,16 @@ private Method getMethod(final Class scope, final String methodName, final Class
...
+                if (methodCls == PythonScopeJpyImpl.CallableWrapper.class || methodCls == groovy.lang.Closure.class) {
```
Comment (kosak) [inline DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:367]: Here and below, can we move this test into a very simple helper method?

So here we can say

```
if (isImplicitCallScope(methodCls)) {
...
```

and then (somewhere)

```
private static boolean isImplicitCallScope(final Class cls) {
  return cls == PythonScopeJpyImpl.CallableWrapper.class || cls == groovy.lang.Closure.class;
}
```

And you can choose whatever name you like for `isImplicitCallScope`

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java [hunk 3]
```diff
@@ -1525,8 +1526,9 @@ public Class visit(MethodCallExpr n, VisitArgs printer) {
 
         expressions=convertParameters(method, argumentTypes, expressionTypes, parameterizedTypes, expressions);
 
-        if (method.getDeclaringClass() == PythonScopeJpyImpl.CallableWrapper.class) {
-            if (scope == null) { // python func call
+        Class methodCls = method.getDeclaringClass();
```
Comment (kosak) [inline DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:1529]: final Class methodCls    ... :-)

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java [hunk 4]
```diff
@@ -1525,8 +1526,9 @@ public Class visit(MethodCallExpr n, VisitArgs printer) {
...
+        if (methodCls == PythonScopeJpyImpl.CallableWrapper.class || methodCls == groovy.lang.Closure.class) {
```
Comment (kosak) [inline DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:1530]: if (isImplicitCallScope(methodCls)) {

**Review by kosak — APPROVED** [review]
I propose making the below changes.
Otherwise, looks good, thank you!

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java (hunk 1 above)
Comment (jmao-denver) [reply DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:366]: done

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java (hunk 3 above)
Comment (jmao-denver) [reply DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:1529]: done

File: DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java (hunk 2 above)
Comment (jmao-denver) [reply DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java:367]: excellent idea, easy refactoring, i think isImplicitCall is a more fitting name

**Pull request merged**